kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/sink/commit/CommitPolicyConditions.scala

```diff
@@ -15,28 +15,31 @@
  */
 package io.lenses.streamreactor.connect.aws.s3.sink.commit
 
+import cats.implicits.toShow
 import com.typesafe.scalalogging.LazyLogging
+import io.lenses.streamreactor.connect.aws.s3.sink.commit.Interval.formatter
 
 import java.time.Clock
 import java.time.Duration
 import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
 import scala.concurrent.duration.FiniteDuration
 import scala.jdk.DurationConverters.ScalaDurationOps
 
-case class ConditionCommitResult(commitTriggered: Boolean, logLine: Option[String])
+case class ConditionCommitResult(commitTriggered: Boolean)
 
-trait CommitPolicyCondition {
-  def eval(context: CommitContext, debugEnabled: Boolean): ConditionCommitResult
+trait CommitPolicyCondition extends LazyLogging {
+  def eval(context: CommitContext): ConditionCommitResult
 }
 
 case class FileSize(maxFileSize: Long) extends CommitPolicyCondition with LazyLogging {
-  override def eval(context: CommitContext, debugEnabled: Boolean): ConditionCommitResult = {
+
+  override def eval(context: CommitContext): ConditionCommitResult = {
     val cond = context.fileSize >= maxFileSize
-    val logLine = Option.when(debugEnabled) {
-      val flushing = if (cond) "*" else ""
-      s"fileSize$flushing: '${context.fileSize}/$maxFileSize'"
-    }
-    ConditionCommitResult(cond, logLine)
+    logger.debug(s"[${context.connectorTaskId.show}] File Size Policy:${context.fileSize}/$maxFileSize.")
```

Comment:
Isn't``` File Size Policy:100/100``` tell the user the flush is triggered ?  When I saw the asterix, I didn't know what it meant.

Reply:
the latest code changes the log in the CommitPolicy when a flush happens and contains the policies state. From that log and the info, it should be enough, no?

Reply:
Here's a scenario.  My connector is configured with the following pseudo-KCQL:

```
INSERT INTO bucket1 SELECT * FROM topic1 WITH_FLUSH_COUNT=100 WITH_INTERVAL = 300 WITH_SIZE=400;
INSERT INTO bucket1 SELECT * FROM topic1 WITH_FLUSH_COUNT=200 WITH_INTERVAL = 600 WITH_SIZE=800;
```

I'm grepping through a massive log file because I'm not sure the committing is working properly (sadly this has happened in the past, usually it is just misunderstanding of configuration, which is why this should be as simple to understand as possible).  There's a lot of noise if I look at every row written by the class `io.lenses.streamreactor.connect.aws.s3.sink.commit.CommitPolicy` because in debug mode it writes all of the conditions that are triggered and not triggered.

It's just a small thing but the asterisk can be used to filter the log entries so I can see which ones were triggered.


Currently I can do something like this:

```
grep 'io\.lenses\.streamreactor\.connect\.aws\.s3\.sink\.commit\.CommitPolicy.*\*' connect_distributed.log
```
This shows me the entries with the class name that contain the asterisk (escaped).

Is there a similar way to filter out only the triggered log events with the updated log lines?

Reply:
As a user, I don't think it matters what triggered the flushing.

Considering the code in CommitPolicy 

```    if (flush) {
      logger.info(
        "[{}]Flushing '{}' for {topic:'{}', partition:{}, offset:{}, {}}",
        context.connectorTaskId.show,
        context.partitionFile,
        context.tpo.topic.value,
        context.tpo.partition,
        context.tpo.offset.value,
        res.flatMap(_.state).mkString(","),
      )
    } else {
      logger.debug(
        "[{}]Not Flushing '{}' for {topic:'{}', partition:{}, offset:{}, {}}",
        context.connectorTaskId.show,
        context.partitionFile,
        context.tpo.topic.value,
        context.tpo.partition,
        context.tpo.offset.value,
        res.flatMap(_.state).mkString(","),
      )
    }
    ```

There is a way to search for `Flushing` to find when it did and, from there, extract the info. 
Another change is to add the connector name and the task to the log, and a user can be specific on which connector name is searching for flushing information and the source topic.

Reply:
All I can say is this tooling was designed to provide that which is necessary to debug the issues we occasionally see with committing. I believe this information is necessary and has actively helped me debug these issues. Not sure there's anything else I can say. Given that this PR's primary focus is to address the byte issues I feel like these changes don't have a business story or requirement behind them and are undoing some work time was already spent on designing and implementing and I'm not seeing any benefit except distraction from other work

Reply:
While the PR should not have addressed the changes on logging,  I do not agree that * is an intuitive way to search for a commit (my first take was that it was a typo), the log did not have the connector name to distinguish where it came from.

Furthermore, the CommitPolicy log code should not depend on the log enabled or not and should store as INFO everytime a flush happens, and the non-flush as debug. In essence, the information is there and improved logging. It has not taken anything from the previous behaviour.

Reply:
This is the story I was trying to achieve...

As an (ops user analysiing logs|developer reading the log file to understand the sequence of events), I would like to be able to see **at a glance** the list of conditions that were evaluated upon a commit, and **at a glance** understand which ones triggered the commit.

To put it into more detail.


As it stands in the PR, we have something like this (simplified):

```
Commit happened!
interval: 5/10
count: 1000000/10000000
size: 30000/30000
```


It committed.  To understand why you have to read the 3 lines below in full, understand the numbers.  Wait, how many zeroes on that count?  Oh, that didn't cause the commit.

Analysing the logs I would love there to be some sort of visual clue that helps understand which conditions caused the trigger without reading all of that.  This was originally implemented as an asterisk.

```
Commit happened!
interval: 5/10
count: 1000000/10000000
size(*): 30000/30000
```

but also what is important to me is it is something for the eye to catch on so that I don't have to read and understand all of the numbers.

In addition, there are also more advanced methods to extract the information from the logs - so I can gather the information about which rules are triggering which commit on masse.  For example, this is a command that I use on the logs to gather statistics how about many times the logs were triggered by which conditions:

```
awk '/(fileSize\*|count\*|interval\*):/ { match($0, /(fileSize\*|count\*|interval\*):/); print substr($0, RSTART, RLENGTH) }' logfile.txt | sort | uniq -c
```

Which prints something like this:

```
   1 count*:
   4 fileSize*:
```

which is operationally useful for finding out which triggers are causing commits (and the inverse will apply too, if you want to find which triggers are not operating without reading the entire log file).

```
awk '/(fileSize|count|interval):/ { match($0, /(fileSize\*|count\*|interval\*):/); print substr($0, RSTART, RLENGTH) }' logfile.txt | sort | uniq -c
```

There's nothing that indicates it has to be an asterisk however this is just something that seemed catching to the eye and a single character.  But IMO making the distinction between active and inactive conditions/triggers is very useful to log analysis.

And that is why I would prefer not to lose this tool.